Engine/src/Engine/Rendering/Renderer.cpp

```diff
@@ -221,6 +226,37 @@ namespace Engine
 			//Draw currently bound objects.
 			glDrawElements(GL_TRIANGLES, vertices.numIndices, GL_UNSIGNED_INT, nullptr);
 		}
+		
+		//Get entities that contain text component.
+		const auto textRenderables = scene.getEntities<const TextComponent>();
+		
+		glUseProgram(m_text.m_textShaderProgram);
+		GLuint mvpID = glGetUniformLocation(m_text.m_textShaderProgram, "mvp");
+
+		for (auto [entity, text] : textRenderables.each())
+		{
+			//Set a default transform component and color if the object does not contain one.
+			TransformComponent transform;
+			glm::vec4 color{ .5f,.5f,.5f,1.f };
+
+			//Change the transform component if the entity contains one.
+			if (scene.m_registry.all_of<TransformComponent>(entity)) { transform = scene.m_registry.get<const TransformComponent>(entity); }
```

Comment:
can we assume this is always the case since an entity with text should also have a transform?

Reply:
Entities with the text component don't necessarily need a transform component. I wanted to remove that dependency given the nature of ECS, so I have a default transform initialized above. This is just on the chance that there is a transform component.